lib/auth.guard.ts

```diff
@@ -60,6 +62,14 @@ function createAuthGuard(type?: string | string[]): Type<CanActivate> {
     }
 
     getRequest<T = any>(context: ExecutionContext): T {
+      if (context.getType<ContextType | 'graphql'>() === 'graphql') {
+        const { GqlExecutionContext } = loadPackage(
+          '@nestjs/graphql',
+          'AuthGuard'
+        );
+        return GqlExecutionContext.create(context).getContext().req;
```

Comment:
What if someone has a different value than `req` for the context? Like with Fastify the property is `request`.

Reply:
Good catch! Didn't know that Fastify adapter uses different properties. Right now i don't have any good ideas on how to sole this issue. Also there are some other adapters, and their properties may be different too

Reply:
I don't think this depends on platform. Doesn't `getContext()` return the context provided to the `GraphQLModule`/`ApolloServer` constructor ([ref](https://www.apollographql.com/docs/apollo-server/data/resolvers/#the-context-argument))? So it would depend on how people initialize the context, e.g.
```ts
GraphQLModule.forRoot({
  context: ({ req }) => {
    return {
      foo: 'bar',
      contextRequest: req,
    }
  }
})
```

Maybe a better solution is to make this more first class; e.g.

https://github.com/nestjs/graphql/blob/f80f28770db48848160ddf82a81547f537663a5e/lib/services/gql-execution-context.ts#L8-L19

Maybe override the constructor to store the request
```ts
class GqlExecutionContext {
  constructor(readonly private originalContext: ExecutionContext, ...) {
    super(...)
  }

  static create(context: ExecutionContext) {
    ...
    new GqlExecutionContext(context, ...)
  }

  getRequest<T = any>() {
    return this.originalContext.switchToHttp().getRequest()
  }
}
```
or something?

Reply:
From the same docs you linked
> The fields of the object passed to your context function differ if you're using middleware besides Express. [See the API reference for details.](https://www.apollographql.com/docs/apollo-server/api/apollo-server/#apolloserver).

Reply:
:+1: still, this assumes that someone set up the `context` key correctly and wouldn't account for someone using the snippet I posted above. Nothing wrong with it; it would just require documentation (admittedly, the workaround we have to do because of this problem should have also been documented)

Reply:
[There is current documentation on using Passport with GraphQL](https://docs.nestjs.com/techniques/authentication#graphql), though it still is also only valid for working with Express (as at the time of writing the docs, `apollo-server-fastify` wasn't compatible). Now when setting the context for fastify, the context should be

```ts
context({ request, reply}) => ({ request, reply })
```

The snippet above looks promising, but I would have to check how well `context.switchToHttp().getRequest()` works on a GQL request in the first place, as the arguments for HTTP and GQL are different.

Reply:
@jmcdo29 we should be able to fix this on the `@nestjs/graphql` level, here https://github.com/nestjs/graphql/blob/master/lib/utils/merge-defaults.util.ts#L9